### PR TITLE
[core] [easy] [no-op] Fix comment for scheduling id and apply thread annotation

### PR DESCRIPTION
**Pull request by dentiny**
Changes behavior of StringIdMap::Insert to recognize max_id. This param is only used by unit tests.

### CODE REVIEW

File: src/ray/common/scheduling/scheduling_ids.h [hunk 1]
```diff
@@ -86,6 +81,12 @@ class StringIdMap {
 
   /// Get number of identifiers.
   int64_t Count();
+
+ private:
```
Comment (dentiny) [inline src/ray/common/scheduling/scheduling_ids.h:85]: Follows https://google.github.io/styleguide/cppguide.html#Declaration_Order

File: src/ray/common/scheduling/scheduling_ids.h [hunk 2]
```diff
@@ -73,7 +68,7 @@ class StringIdMap {
   ///
   /// \param String ID to be inserted.
   /// \param max_id The number of unique possible ids. This is used
-  ///               to force collisions for testing. If -1, it is not used.
+  ///               to force collisions for testing. If 0, it is not used.
```
Comment (dentiny) [inline src/ray/common/scheduling/scheduling_ids.h:71]: Incorrect comment.

File: src/ray/common/scheduling/scheduling_ids.h [hunk 3]
```diff
@@ -27,7 +27,7 @@
 #include "ray/util/util.h"
 
 /// Limit the ID range to test for collisions.
-#define MAX_ID_TEST 8
```
Comment (dentiny) [inline src/ray/common/scheduling/scheduling_ids.h:30]: Follows https://google.github.io/styleguide/cppguide.html#Constant_Names

File: src/ray/common/scheduling/scheduling_ids.cc [hunk 4]
```diff
@@ -44,7 +43,7 @@ int64_t StringIdMap::Insert(const std::string &string_id, uint8_t max_id) {
   if (sit == string_to_int_.end()) {
     int64_t id = hasher_(string_id);
     if (max_id != 0) {
-      id = id % MAX_ID_TEST;
```
Comment (rynewang) [inline src/ray/common/scheduling/scheduling_ids.cc:47]: Q: should this actually be 

    id = id % max_id

like in line 58?

Comment (dentiny) [reply src/ray/common/scheduling/scheduling_ids.cc:47]: I feel like it's intended, so we could pass in `max_id` = 0 in tests, meanwhile still capping the max id.

Comment (rynewang) [reply src/ray/common/scheduling/scheduling_ids.cc:47]: The only place that used this Insert is

https://github.com/ray-project/ray/blob/e4b9561d07b9b69e869955959886ec88d71e5efb/src/ray/raylet/scheduling/cluster_resource_scheduler_test.cc#L226

so I wonder if we can just remove this kMaxIdForTest

Comment (dentiny) [reply src/ray/common/scheduling/scheduling_ids.cc:47]: Removed.

**Comment by dentiny** [comment]
@rynewang could you please help me approve / merge it? thanks!

**Review by rynewang — APPROVED** [review]

**Pull request merged**